### DISCUSSION TITLE
Docs are missing info on how to enable inline errors(configure pyright) (at least in VSCode they are not automatic)

### DISCUSSION THREAD

**Discussion by toinbis** [fastapi/sqlmodel#1924] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
"""Main module"""
from typing import Optional

from sqlmodel import Field, SQLModel


class Hero(SQLModel, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
    name: str
    secret_name: str
    age: Optional[int] = None


hero_1 = Hero(name="Deadpond", secret_name="Dive Wilson")
hero_2 = Hero(name="Spider-Boy", secret_name="Pedro Parqueador")
hero_3 = Hero(name="Rusty-Man", secret_name='Tom')

a = hero_1.name + 5
```

### Description

https://github.com/tiangolo/sqlmodel/blob/main/docs/index.md#editor-support suggests that inline errors will be automatic. At least in VSCode, they are not. Docs need clarification on how to achieve this.

### Wanted Solution

Docs should note that additional vscode config is needed to get inline errors.

### Wanted Code

```python
"python.analysis.diagnosticSeverityOverrides": {
        "reportGeneralTypeIssues": "information"
    }
```
```


### Alternatives

_No response_

### Operating System

macOS

### Operating System Details

_No response_

### SQLModel Version

0.0.4

### Python Version

3.9.0

### Additional Context

This is the default result you get using sqlmodel in vscode:
![Screenshot 2021-08-31 at 14 43 26](https://user-images.githubusercontent.com/68685/131496606-c6a5f232-af59-440c-920b-ab29b8057832.png) . And it's different from what the docs state.

**Comment by toinbis** [comment]
The issue ended up being missformated. Here is the screenshot that was expected to be attached:

![Screenshot 2021-08-31 at 14 43 26](https://user-images.githubusercontent.com/68685/131497107-6c25fbe3-f836-4a8c-abeb-891414272cac.png)

**Comment by faresbakhit** [comment]
Wouldn't it be better to just advice enabling all type checking errors at `basic` mode instead of `reportGeneralTypeIssues` only?
```json
"python.analysis.typeCheckingMode": "basic"
```
In my experience, other type checking errors can be very helpful too.

**Comment by toinbis** [comment]
@FaresAhmedb - sure, `"reportGeneralTypeIssues": "information"` was just the first setting that allowed me to replicate the behvaiour from screenshot presented in docs. My main point was that additional elaboration in when `inline errors` appear, probably customized for each and every IDE/editor. Which particular setting to advise using - I have no competence to advise :)

**Comment by tiangolo** [comment, marked as answer]
I'm coming years later to this :sweat_smile: 

But in case anyone is still struggling, SQLModel uses modern standard types, including `dataclass_transform`. So, any editor that supports modern types would be able to work with it and provide inline errors and autocompletion.

In most cases, the issues with editors come from them not being able to detect the specific Python virtual environment used, or having an incorrect one configured. And then not being able to parse/import and use the source for SQLModel to understand its types.

Using uv would help standardize the way venvs are found, and then checking in the docs for the specific editor about selecting the Python environment to be used.
